windlass/pins.py

```diff
@@ -79,11 +79,13 @@ def iter_pin_files(self, repodir=None):
             fnlist = [i for i in windlass.tools.all_blob_names(repodir.tree)]
             for pin_files_glob in pin_files_globs:
                 for blob_name in fnmatch.filter(fnlist, pin_files_glob):
-                    yield (repodir.tree / blob_name).data_stream
+                    with (repodir.tree / blob_name).data_stream as stream:
+                        yield stream
         else:
             for pin_files_glob in pin_files_globs:
                 for pin_file in glob.glob(pin_files_glob):
-                    yield open(pin_file)
+                    with open(pin_file) as fp:
+                        yield fp
```

Comment:
When would file handle be closed?

Reply:
As soon as the returned `fp` goes out of scope in the caller, it'll be closed. Would you prefer if I made the caller explicitly close these?

Reply:
No that is better way, but then we should remove call somewhere else.